community/neo4j/src/test/java/org/neo4j/index/backup/IndexBackupIT.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.index.backup;
+
+import org.apache.commons.io.FilenameUtils;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.impl.api.index.IndexingService;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointer;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.SimpleTriggerInfo;
+import org.neo4j.test.rule.EmbeddedDatabaseRule;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class IndexBackupIT
+{
+    @Rule
+    public EmbeddedDatabaseRule database = new EmbeddedDatabaseRule( getClass() ).startLazily();
+
+    @Test
+    public void concurrentIndexSnapshotUseDifferentSnapshots() throws Exception
+    {
+        Label label = Label.label( "testLabel" );
+        prepareDatabase( label );
+
+        CheckPointer checkPointer = resolveDependency( CheckPointer.class );
+        IndexingService indexingService = resolveDependency( IndexingService.class );
+
+        forceCheckpoint( checkPointer );
+        ResourceIterator<File> firstCheckpointSnapshot = indexingService.snapshotStoreFiles();
+        generateData( label );
```

Comment:
How can we enhance this test so that we know that the changes result in lots of Lucene file changes, at least some added, some changed and some deleted?